purrr_basics/purrr_basics.Rmd

```diff
@@ -126,59 +129,366 @@ gapminder[1:6, "pop"]
+  gapminder, 
+  across(where(is.numeric), list(mean = mean, median = median)) #<<
+)
+```
+
+]
+
+
+---
+
+## Review: `tidyselect`
+
+### 'Behind the scenes' workhorse for `dplyr::select()`,  `dplyr::pull()`, and `tidyr::pivot_` functions
+
+### `starts_with()`, `ends_with()`, `contains()`, `matches()`, etc.
+
+--
+
+.pull-left[
+
+.small[
+
+```{r slide-21-left-tidyselect}
+# column names contain a word
+select(diabetes, ends_with("ht"))
+```
+
+]
+
+]
+
+--
+
+.pull-right[
+
+.small[
+
+```{r slide-21-right-tidyselect}
+# regular expression
+select(diabetes, matches("\\d")) 
+```
+
+]
+
+]
+
+---
+
+## `mutate(across())` & `summarise()`
+
+### Perform group-wise summaries with `mutate(across())`, `group_by()` and `summarise()`
+
+```{r slide-22-mutate-across-summarise-concept-map, eval=FALSE}
+<DATA> %>% 
+  group_by(<GROUPING VAR>) #<<
+  summarise(across(c(<VARIABLES>), list(<NAMES> = <FUNCTIONS>))) 
+```
+
+---
+
+## `mutate(across())` & `summarise()`
+
+### Combine with `tidyselect()` helpers
+
+```{r slide-23-mutate-across-tidyselect-summarise-concept-map, eval=FALSE}
+<DATA> %>% 
+  group_by(<GROUPING VAR>) 
+  summarise(across(contains(<VARIABLES>), list(<NAMES> = #<<
+      <FUNCTIONS>))) 
+```
+
+---
+
+## `mutate(across())` & `summarise()`
+
+```{r slide-24-code, eval=FALSE}
+gapminder %>% 
+  group_by(continent) %>% 
+    summarise(across(c("lifeExp", "gdpPercap"), list(med = median, #<<
+      iqr = IQR))) #<<
+```
+
+--
+
+```{r slide-24-output, echo=FALSE, highlight.output = c(2)}
+gapminder %>% 
+  group_by(continent) %>% 
+    summarise(across(c("lifeExp", "gdpPercap"), list(med = median, 
+      iqr = IQR)))
+```
+
+---
+
+## `mutate(across())` & `summarise()`
+
+
+### New column names are created with `_[function name]`
+
+```{r slide-25-code, eval=FALSE, highlight.output = c(2)}
+gapminder %>% 
+  group_by(continent) %>% 
+    summarise(across(c("lifeExp", "gdpPercap"), list(med = median, #<<
+      iqr = IQR))) #<<
+```
+
+--
+
+```{r slide-25-output, echo=FALSE, highlight.output = c(2)}
+gapminder %>% 
+  group_by(continent) %>% 
+    summarise(across(c("lifeExp", "gdpPercap"), list(med = median, #<<
+      iqr = IQR))) #<<
+```
+
+---
+
+## `mutate(across())` & `summarise()`
+
+### Control output names with `.names` argument 
+
+```{r slide-26-code, eval=FALSE}
+gapminder %>% 
+  group_by(continent) %>% 
+    summarise(across(c("lifeExp", "gdpPercap"), list(med = median, #<<
+      iqr = IQR) .names = "{.fn}_{.col}")) #<<
+```
+
+--
+
+```{r slide-26-output, echo=FALSE, highlight.output = c(2)}
+gapminder %>% 
+  group_by(continent) %>% 
+    summarise(across(c("lifeExp", "gdpPercap"), list(med = median, #<<
+      iqr = IQR), .names = "{.fn}_{.col}")) #<<
+```
+
+---
+
+## `mutate(across())` & `summarise()`
+
+### `tidyselect()` helpers and named columns
+
+```{r slide-27-mutate-across-summarise-code, eval=FALSE}
+diabetes %>% 
+  group_by(gender) %>% 
+    summarise(across(starts_with("bp"), list(mean = mean))) #<<
```

Comment:
Let's just do `mean` for these instead of `list(mean = mean)`.

Reply:
I can do this, but the output isn't very clear: 

```
## # A tibble: 2 × 5
##   gender `1_bp.1s` `1_bp.1d` `1_bp.2s` `1_bp.2d`
##   <chr>      <dbl>     <dbl>     <dbl>     <dbl>
## 1 female      136.      82.5      153.      91.8
## 2 male        138.      84.5      151.      93.5
```

Should I swap `{.fn}` and `{.col}`?